presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,549 @@ Tuning Presto
+
+
+``query.low-memory-killer.delay``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (duration, at least ``5s``)
+ * **Default value:** ``5 m``
+ * **Description:** Delay between cluster running low on memory and invoking killer. When this value is low, there will be instant reaction for running out of memory on cluster. This may cause more queries to fail fast but it will be less often that query will fail in unexpected way.
+
+
+``query.low-memory-killer.enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** This property controls if there should be killer of query triggered when cluster is running out of memory. The strategy of the killer is to drop largest queries first so enabling this option may cause problem with executing large queries in highly loaded cluster but should increase stability of smaller queries.
+
+
+``query.manager-executor-pool-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``5``
+ * **Description:** Size of thread pool used for garbage collecting after queries. Threads from this pool are used to free resources from canceled queries, enforcing memory limits, queries timeouts etc. Higher number of threads will allow to manage memory more efficiently, so it may be increased to avoid out of memory exceptions in some scenarios. On the other hand higher value here may increase CPU usage for garbage collecting and use additional constant memory even if there is nothing to do for all of the threads.
+
+
+``query.min-expire-age``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (duration)
+ * **Default value:** ``15 m``
+ * **Description:** This property describes time after which the query metadata may be removed from server. If value is low, it's possible that client will not be able to receive information about query completion. The value describes minimum time that must pass to remove query (after it's considered completed) but if there is space available in history queue the query data will be kept longer. The size of history queue is defined by ``query.max-history`` property (``100`` by default).
+
+
+``query.max-concurrent-queries``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``1000``
+ * **Description:** **Deprecated** Describes how many queries be processed simultaneously in single cluster node. It shouldn't be used in new configuration, the ``query.queue-config-file`` can be used instead.
```

Comment:
Describes how many queries can be processed simultaneously in a single cluster node. In new configurations, the ``query.queue-config-file`` should be used instead.

Same question about documenting deprecated properties in the tuning guide.